modules/caddyhttp/reverseproxy/selectionpolicies.go

```diff
@@ -824,9 +825,9 @@ func hostByHashing(pool []*Upstream, s string) *Upstream {
 
 // hash calculates a fast hash based on s.
 func hash(s string) uint32 {
-	h := fnv.New32a()
+	h := xxhash.New()
 	_, _ = h.Write([]byte(s))
-	return h.Sum32()
+	return uint32(h.Sum64())
```

Comment:
As I'm unfamiliar with this hash, do we really want to truncate to 32bits? We could change the return signature to uint64 to accommodate the 64-bit hash output.

Reply:
I agree that it would be better to keep it to the original uint64. I have modified it. Thanks!